src/inmanta/server/services/compilerservice.py

```diff
@@ -719,3 +723,63 @@ async def get_compile_queue(self, env: data.Environment) -> List[model.CompileRu
         """
         compiles = await data.Compile.get_next_compiles_for_environment(env.id)
         return [x.to_dto() for x in compiles]
+
+    @protocol.handle(methods_v2.get_compile_reports, env="tid")
+    async def get_compile_reports(
+        self,
+        env: data.Environment,
+        limit: Optional[int] = None,
+        first_id: Optional[uuid.UUID] = None,
+        last_id: Optional[uuid.UUID] = None,
+        start: Optional[datetime.datetime] = None,
+        end: Optional[datetime.datetime] = None,
+        filter: Optional[Dict[str, List[str]]] = None,
+        sort: str = "requested.desc",
+    ) -> ReturnValue[List[model.CompileReport]]:
+
+        if limit is None:
+            limit = APILIMIT
+        elif limit > APILIMIT:
+            raise BadRequest(f"limit parameter can not exceed {APILIMIT}, got {limit}.")
```

Comment:
This code is present in many API endpoints. Maybe we should move this check into `protocol` at some point. @wouterdb That could be part of a refactoring ticket as well.